### ISSUE TITLE
[49c80ce] Segmentation fault trying to decode an AAC file

### ISSUE THREAD

**Issue by marcelpaulo** [HessiJames/soundkonverter#24]
When I try to convert an AAC file to MP3, SK crashes with segmentation fault, before actually starting to decode the file. Here's the window that pops up:
![soundkonverter the kde crash handler_001](https://cloud.githubusercontent.com/assets/10074769/7666993/88f7228c-fbd0-11e4-8844-1e1c5a269955.jpg)

I built SK from source cloning master at commit 49c80cedcc336a96b45adfefa75f35688df63027. `kdelibs5-data` is at version **4:4.14.6-4~ubuntu3.1**, `faad` at **2.7-8**. I'm running `Xubuntu` **15.04**.

I gathered all info that I thought might be useful, let me know if you need anything else:

- Ran SK on console as `soundkonverter &> >(tee console.log)`, and here's [console.log](http://paste.ubuntu.com/11169003/)
- I configured SK to write the log files to disc, and here's the [log](http://paste.ubuntu.com/11169017/)
- Here's my *profiles.xml*:
```
<!DOCTYPE soundkonverter_profilelist>
<soundkonverter type="profilelist">
 <conversionOptions profile="User defined" profileName="mp3 extreme" pluginName="lame" codecName="mp3">
  <encodingOptions qualityMode="0" bitrate="160" compressionLevel="2" bitrateMode="0" quality="5" cmdArguments="--id3v2-only">
   <data presetFast="0" presetBitrate="192" presetBitrateCbr="0" preset="2"/>
  </encodingOptions>
  <outputOptions outputFilesystem="" outputDirectoryMode="1" outputDirectory=""/>
  <features replaygain="0"/>
 </conversionOptions>
 <conversionOptions profile="User defined" profileName="mp3 insane" pluginName="lame" codecName="mp3">
  <encodingOptions qualityMode="0" bitrate="160" compressionLevel="2" bitrateMode="0" quality="5" cmdArguments="--id3v2-only">
   <data presetFast="0" presetBitrate="192" presetBitrateCbr="0" preset="3"/>
  </encodingOptions>
  <outputOptions outputFilesystem="" outputDirectoryMode="1" outputDirectory=""/>
  <features replaygain="0"/>
 </conversionOptions>
 <conversionOptions profile="User defined" profileName="soundkonverter_last_used" pluginName="lame" codecName="mp3">
  <encodingOptions qualityMode="0" bitrate="160" compressionLevel="2" bitrateMode="0" quality="5" cmdArguments="--id3v2-only">
   <data presetFast="0" presetBitrate="192" presetBitrateCbr="0" preset="2"/>
  </encodingOptions>
  <outputOptions outputFilesystem="" outputDirectoryMode="1" outputDirectory=""/>
  <features replaygain="0"/>
 </conversionOptions>
</soundkonverter>
```
- Here's my [soundkoverterrc](http://paste.ubuntu.com/11169096/)
- And finally, here's the [AAC file](https://docs.google.com/file/d/0B6eQjhXV-AquVlJZWGpaQ3Nock0/edit?usp=drivesdk)

I should mention: I'm able to decode the AAC file calling `faad` directly:
```
paulo@monk:~/stage/edit$ faad 20150516.aac 
 *********** Ahead Software MPEG-4 AAC Decoder V2.7 ******************

 Build: May  1 2012
 Copyright 2002-2004: Ahead Software AG
 http://www.audiocoding.com
 Floating point version

 This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License.

 **************************************************************************

20150516.aac file info:
ADTS, 31.208 sec, 34 kbps, 44100 Hz

  ---------------------
 | Config:  2 Ch       |
  ---------------------
 | Ch |    Position    |
  ---------------------
 | 00 | Left front     |
 | 01 | Right front    |
  ---------------------

Decoding 20150516.aac took:  0.08 sec. 407.36x real-time.
```
SK seems to crash before calling `faad`.

Please let me know how I can help debug this further.

**Comment by HessiJames** [comment]
Thanks, fixed it with commit ced492f.

**Comment by marcelpaulo** [comment]
Thank you so much, @HessiJames, for your speed-lightning fix. I have just tested it and can confirm that the issue has been fixed.
